src/main/java/de/tum/in/www1/artemis/service/RepositoryService.java

```diff
@@ -82,6 +87,53 @@ public RepositoryService(GitService gitService, AuthorizationCheckService authCh
         return fileInBytes;
     }
 
+    /**
+     * Gets the files of the repository and checks whether they were changed during a student participation.
+     * Compares the files from the students repository with the files of the template repository.
+     *
+     * @param repository the students repository with possibly new files and changed files
+     * @param templateRepository the template repository with default files on which the student started working on
+     * @return a map of files with the information if they were changed/are new.
+     */
+    public Map<String, Boolean> getFilesWithInformationAboutChange(Repository repository, Repository templateRepository) {
+        Map<String, Boolean> filesWithInformationAboutChange = new HashMap<>();
+
+        var repoFiles = gitService.listFilesAndFolders(repository).entrySet().stream().filter(entry -> entry.getValue() == FileType.FILE).map(Map.Entry::getKey)
+                .collect(Collectors.toList());
+        var templateRepoFiles = gitService.listFilesAndFolders(templateRepository).entrySet().stream().filter(entry -> entry.getValue() == FileType.FILE).map(Map.Entry::getKey)
+                .collect(Collectors.toList());
+        // Used to avoid going through the loop when the file which the student added was not in the template repository
+        var templateRepoFilesNames = templateRepoFiles.stream().map(File::toString).collect(Collectors.toList());
+
+        repoFiles.forEach(file -> {
+            var fileName = file.toString();
+            // When template repository does not contain this file it is new/changed, no need to iterate through the loop
+            if (!templateRepoFilesNames.contains(fileName)) {
+                filesWithInformationAboutChange.put(fileName, true);
+            }
+            else {
+                templateRepoFiles.forEach(templateFile -> {
+                    try {
+                        // Check to assure we are comparing the correct files
+                        if (file.toString().equalsIgnoreCase(templateFile.toString())) {
+                            // When having the same content, it was not changed
+                            if (FileUtils.contentEquals(file, templateFile)) {
+                                filesWithInformationAboutChange.put(fileName, false);
+                            }
+                            else {
+                                filesWithInformationAboutChange.put(fileName, true);
+                            }
+                        }
+                    }
```

Comment:
Why do we need this loop? We already verified that the file is in that list, so I would argue that it can be deleted.
Consider replacing `templateRepoFiles` and `templateRepoFilesNames` with a `Map<String, File>` instead?

Reply:
Thank you, using a `Map<String, File>` was indeed better! Done in `28d558`